pkg/controllers/jobset_controller.go

```diff
@@ -377,15 +426,47 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 			return err
 		}
 	}
-
 	var lock sync.Mutex
 	var finalErrs []error
-
-	for _, rjob := range js.Spec.ReplicatedJobs {
-		jobs, err := constructJobsFromTemplate(js, &rjob, ownedJobs)
+	allJobsCreatedForStartupPolicy := false
+	statusOfFirstJob := findReplicatedStatus(replicatedJobStatus, js.Spec.ReplicatedJobs[0].Name)
+	if !isJobSetSuspended(js) && inOrderStartupPolicy(js.Spec.StartupPolicy) && !replicatedJobsAtLeastReady(js.Spec.ReplicatedJobs[0].Replicas, statusOfFirstJob) {
+		err := r.ensureCondition(ctx, js, corev1.EventTypeNormal, true, generateStartupPolicyCondition(false))
```

Comment:
@ahg-g So I made a little adjustment to KEP here. I set the condition for startup policy on the first entry in this list. This notifies the condition/user that startup policy is being applied and sets the condition to false.
And once the startup policy is complete I move it back to true.

I decided to go with emitting events as I had some issues with duplicate conditions. It is also quite noisy for updates so wasn't sure if we should be updating conditions/events on every loop of create jobs.

Reply:
Lets do the events as a followup, just to simplify the review and focus on getting the logic right.

As for the condition, we already check that if the condition doesn't change, we don't send an update: https://github.com/kubernetes-sigs/jobset/blob/6f9671ef94a6b8e4c9f389e8e39dff3a0053b38c/pkg/controllers/jobset_controller.go#L538

Reply:
I updated some of that code for this PR actually. It was probably a bug I introduced.

I discovered that we were never checking the message so we would only look at reason to determine unique conditions so when I changed the message I was not getting the new condition. 

I also had to update the condition to force an update on false. We would not updates conditions if they were first false (only true -> false) it seemed. Didn't make much sense for startup policy.